android/src/main/java/com/aerobotics/DjiMobile/DJIMobile.java

```diff
@@ -894,26 +924,62 @@ public void onValueChange(@Nullable Object oldValue, @Nullable Object newValue)
 
   private void startUltrasonicHeightListener() {
     startEventListener(SDKEvent.AircraftUltrasonicHeight, new EventListener() {
-          @Override
-          public void onValueChange(@Nullable Object oldValue, @Nullable Object newValue) {
-      if (newValue != null && newValue instanceof Float) {
-        sendEvent(SDKEvent.AircraftUltrasonicHeight, newValue);
+      @Override
+      public void onValueChange(@Nullable Object oldValue, @Nullable Object newValue) {
+        if (newValue != null && newValue instanceof Float) {
+          sendEvent(SDKEvent.AircraftUltrasonicHeight, newValue);
+        }
       }
-          }
     });
   }
 
+  private void startRemoteControllerFlightModeListener() {
+    getInitialSDKEventValue(SDKEvent.RemoteControllerFlightMode, new GetCallbackWrapper() {
+      @Override
+      public void onSuccess(@NonNull Object o) {
+        handleRemoteControllerFlightModeUpdate(o, true);
+      }
+    });
+    startEventListener(SDKEvent.RemoteControllerFlightMode, new EventListener() {
+      @Override
+      public void onValueChange(@Nullable Object oldValue, @Nullable Object newValue) {
+        handleRemoteControllerFlightModeUpdate(newValue, false);
+      }
+    });
+  }
+
+  private void handleRemoteControllerFlightModeUpdate(@Nullable Object newValue, Boolean realtime) {
+    if (newValue != null && newValue instanceof RemoteControllerFlightMode) {
+      this.eventSender.processEvent(
+        SDKEvent.RemoteControllerFlightMode,
+        ((RemoteControllerFlightMode)newValue).name(),
+        realtime
+      );
+    }
+  }
+
   private void startCompassHasErrorListener() {
+    getInitialSDKEventValue(SDKEvent.CompassHasError, new GetCallbackWrapper() {
+      @Override
+      public void onSuccess(@NonNull Object o) {
+        handleCompassHasErrorUpdate(o, true);
+      }
+    });
+
```

Comment:
@MikeWoots I've abstracted the getting of the initial value and the error handling of that in the `GetCallbackWrapper` interface and `getInitialSDKEventValue(...)` method